### PR TITLE
feat(rdc): Define a few reset scnearios

### PR DESCRIPTION
**Pull request by eunchan**
This commit defines scenarios of reset manager's sw controlled reset
signals and SPI_DEVICE FIFO reset signals.

### CODE REVIEW

File: hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl [hunk 1]
```diff
@@ -4,8 +4,63 @@
...
+###############
+# Reset Group #
+###############
+
+# List of external reset group
+set ext_rst_lst {}
+# POR_N active low
+lappend ext_rst_lst [list { POR_N } {reset low}]
+
+foreach_in_collection j [get_pins top_earlgrey.rstmgr_aon_resets.*] {
```
Comment (tjaychen) [inline hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl:16]: o wow, do we need to do this to inform the tool where all the resets are?

File: hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl [hunk 2]
```diff
@@ -4,8 +4,63 @@
...
+    lappend ext_rst_lst [list [get_attribute ${j} full_name] {reset low}]
+}
+
+set_reset_group $ext_rst_lst -name ext_rst
```
Comment (tjaychen) [inline hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl:20]: does this imply they can all be asserted independently? if it's like "clock groups", @jeoongp or i can help to make this more intelligent

File: hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl [hunk 3]
```diff
@@ -4,8 +4,63 @@
...
 ###################
 
 # TODO: SPI_CSB asserted(release from reset) only when SYS_RST_N is not asserted
+
+
+# POR_N
+set_reset_scenario { {POR_N {reset { @t0 0 } { #10 1}}}} -name ScnPOR
+
+# AST POK
+
```
Comment (tjaychen) [inline hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl:33]: do we want me to point out what these are?

File: hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl (hunk 1 above)
Comment (eunchan) [reply hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl:16]: The tool figured out the POR_N and SW reset signals (+ SPI_DEVICE FIFO resets). I need to check if these are all resets or not.

Comment (tjaychen) [reply hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl:16]: o those definitely are.

File: hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl (hunk 2 above)
Comment (eunchan) [reply hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl:20]: It seems to be a little bit more complex than that. The example shows the reset group is more like relationship between the resets. For example, one reset signal could belongs to multiple reset groups.

File: hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl (hunk 3 above)
Comment (eunchan) [reply hw/top_earlgrey/rdc/chip_earlgrey_asic_scenario.tcl:33]: Yep. Would be great to create a simple scenario (assertion -> deassertion) for POK signals generating from AST.

**Review by tjaychen — APPROVED** [review]

**Pull request merged**